src/App.tsx

```diff
@@ -1,24 +1,226 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback, useEffect, useState,
+} from 'react';
+import {
+  addTodo, deleteTodo, getTodos, updateTodo,
+} from './api/todos';
+import { FilterType } from './types/FilterType';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Errors } from './components/Errors';
+import { ErrorMessage } from './types/ErrorMessage';
 
-const USER_ID = 0;
+const USER_ID = 6713;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.ALL);
+  const [errorMessage, setErrorMessage]
+    = useState<ErrorMessage>(ErrorMessage.NONE);
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+
+  const activeTodosLength = todos.filter((todo) => !todo.completed).length;
```

Comment:
apply useMemo here to avoid unnecessary calculations on each render

Reply:
done